src/nodes/accessors/Normal.js

```diff
@@ -69,16 +69,20 @@ export const transformNormal = /*@__PURE__*/ Fn( ( [ normal, matrix = modelWorld
 
 } );
 
-export const transformNormalToView = /*@__PURE__*/ Fn( ( [ normal, matrix = modelWorldMatrix ], builder ) => {
+export const transformNormalToView = /*@__PURE__*/ Fn( ( [ normal ], builder ) => {
 
-	const modelNormalMatrix = builder.renderer.nodes.modelNormalMatrix;
+	const modelNormalViewMatrix = builder.renderer.nodes.modelNormalViewMatrix;
 
-	if ( modelNormalMatrix !== null ) {
+	if ( modelNormalViewMatrix !== null ) {
 
-		return modelNormalMatrix.transformDirection( normal );
+		return modelNormalViewMatrix.transformDirection( normal );
 
 	}
 
-	return cameraViewMatrix.transformDirection( transformNormal( normal, matrix ) );
+	//
+
+	const transformedNormal = modelNormalMatrix.mul( normal );
+
+	return cameraViewMatrix.transformDirection( transformedNormal );
```

Comment:
What is the reason you are using `transformDirection()` here?

That method is designed to be applied to vectors such as the tangent or bi-tangent, not the normal. We use the `normalMatrix` when transforming normal vectors.

Using `transformDirection()` will be correct only if the camera has uniform scale (the typical use case), but what is the point of doing it this way?

What is it about this approach that is making the code faster?

Reply:
`mesh.normalMatrix` is a view space matrix, this should be called `normalViewMatrix` as I updated in TSL.

`modelNormalMatrix.mul( normal )` will convert the normals to world space, and `transformDirection()` to view space. I could use `mul().normalize()` instead of `transformDirection()` but I don't see much advantage in that.

> What is it about this approach that is making the code faster?

This doesn't use CPU-computed viewMatrixes, so we don't need to update GPU buffers every frame since the global position doesn't always change.

Reply:
What is your justification for being able to use `transformDirection()`, instead of the normal matrix, to transform the normal from world space to view space?

Reply:
`modelNormalMatrix` is not the same as `mesh.normalMatrix`.

I used `matrix3.getNormalMatrix( object.matrixWorld )` instead of `matrix3.getNormalMatrix( object.modelViewMatrix )`, this is a normal world space matrix.

`mesh.normalMatrix` is a view space matrix and should be have the suffix `NormalViewMatrix` in TSL context, it is not used in this code, you can access it using `highPrecisionModelNormalViewMatrix`, with the approach mentioned in https://github.com/mrdoob/three.js/pull/29299#issue-2501183898 updated.

I believe the following explanation would be the same as the one mentioned above.

Reply:
The view matrix is assumed to be orthogonal and have uniform scale. Plus, we normalize the result after applying the matrix transform.

Consequently, normal vectors can be transformed with the same matrix as direction vectors.

Thus, it is OK to use `transformDirection()`, instead of the normal matrix, to transform the normal from world space to view space.

Reply:
tldr; This looks OK to me.